modules/channel-vonage/src/backend/client.ts

```diff
@@ -296,9 +298,21 @@ export class VonageClient {
 
       if (picture) {
         await this.sendImage(event, { url: picture, type: 'image' })
+
+        // Sandbox API is limited to one call per second
+        if (this.config.useTestingApi === true) {
+          await Promise.delay(1000)
+        }
```

Comment:
can you hide this logic in the sendImage function

Reply:
If I do this, we are going to wait 1sec during the typing event and one second after sending an image. I know this has no side effects other than wasting CPU, but since the only use case where we want to wait one second between calls to sendMessage is with down rendering, I think it makes more sense if this logic stays here.

Reply:
Oh didn't get it. Thanks for clarifying

Reply:
Is only the image sending limited by this ? I would think that any type of request has the same limitations, no?

Reply:
@allardy 

The reason why we only wait in this particular case and not each time we send images is due to the fact that when we down render the carousel, we potentially send multiple messages (images and text) at once. 

Since the sandbox is limited to 1 call per second, we have to make sure sending a carousel does not bust Vonage's rate limiter by waiting 1 second between each calls to sendMessage.

Reply:
@EFF @allardy I made changes so that we wait 1 sec after each call to sendMessage. This way, we don't have to handle each edge case separately.